src/widgets/LineChart.js

```diff
@@ -0,0 +1,115 @@
+import { APP_FONT_FAMILY, GREY, LIGHT_GREY, DARK_GREY, SUSSOL_ORANGE } from '../globalStyles';
+
+export const LineChart = ({ title, type, data }) => {
+  const [dimensions, setDimensions] = useState({ height: null, width: null });
+  const { height, width } = dimensions;
+
+  // Victory Native sizes are set using absolute values. Parents dimensions are used to
+  // calculate relative values for width and height for each chart.
+  const onLayout = event => {
+    const newDimensionsObj = {
+      height: event.nativeEvent.layout.width,
+      width: event.nativeEvent.layout.height,
+    };
+    setDimensions(newDimensionsObj);
+  };
+  // X-axis for bar and line charts.
+  const renderXAxis = () => {
+    const tickTruncate = label => (label.length > 11 ? `${label.slice(0, 11)}...` : label);
+    return <VictoryAxis tickFormat={tickTruncate} style={victoryStyles.axisX.style} />;
+  };
+
+  // Y-axis for bar and line charts.
+  const renderYAxis = () => <VictoryAxis dependentAxis style={victoryStyles.axisY.style} />;
+
+  const {
+    padTop,
+    padBottom,
+    padLeft,
+    padRight,
+    dotSize,
+    dotStyle,
+    lineStyle,
+  } = victoryStyles.lineChart;
+
+  const padding = {
+    top: height * padTop,
+    bottom: height * padBottom,
+    left: width * padLeft,
+    right: width * padRight,
+  };
+  return (
+    <View style={localStyles.ChartContainer} onLayout={onLayout}>
+      <VictoryChart width={width} height={height} padding={padding}>
+        <VictoryScatter size={dotSize} style={dotStyle} data={data} />
+        <VictoryLine style={lineStyle} data={data} />
+        {renderXAxis()}
+        {renderYAxis()}
+      </VictoryChart>
+    </View>
+  );
+};
+
+LineChart.propTypes = {
+  title: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  data: PropTypes.array.isRequired,
+};
+
+const localStyles = StyleSheet.create({
+  ChartContainer: {
+    width: '75%',
+    minHeight: '100%',
+    backgroundColor: 'white',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+});
+
+const victoryStyles = {
+  axisX: {
+    fixLabelOverlap: true,
+    style: {
+      axis: { stroke: LIGHT_GREY },
+      ticks: { stroke: DARK_GREY },
+      tickLabels: { fontFamily: APP_FONT_FAMILY, fill: GREY, textAnchor: 'end', angle: -45 },
+    },
+  },
+  axisY: {
+    fixLabelOverlap: false,
+    style: {
+      axis: { stroke: LIGHT_GREY },
+      ticks: { stroke: DARK_GREY },
+      tickLabels: { fontFamily: APP_FONT_FAMILY, fill: GREY },
+    },
+  },
+  barChart: {
+    padTop: 0.1,
+    padBottom: 0.15,
+    padLeft: 0.1,
+    padRight: 0.05,
+    padDomain: 0.05,
+    style: { data: { fill: SUSSOL_ORANGE } },
+  },
+  lineChart: {
+    padTop: 0.1,
+    padBottom: 0.15,
+    padLeft: 0.1,
+    padRight: 0.05,
+    dotSize: 3.5,
+    dotStyle: { data: { fill: SUSSOL_ORANGE } },
+    lineStyle: { data: { stroke: SUSSOL_ORANGE } },
+  },
+};
```

Comment:
not sure if this was the same in the last chart also, but a few of these fields aren't used in this component!

For example, `axisX.fixLabelOverlap` isn't used.. which means there is just a `style` field.. so we have some un-needed nesting imo.

Either we can tidy these up for each file, or create a single file for each chart component to import from. I think just tidying it up in each file is the tidiest?

Reply:
Yes, good eye btw. I'm checking it right now.
e.g. `fixLabelOverlap: false` is the default value, so no needed and also `pieChart` styles doesn't belong to this component, I removed them as well.
I just checked the recently merged `PieChart` component and it looks good.

By now I guess tidying it up in each file is better and later we can think about moving these styles to added files imo. Can add an issue to do it too!

Reply:
perfect!

Reply:
Thanks! Done here 0103d70
I will definitely add the issue related with these styles and work on polish them.